### ISSUE TITLE
Updating G3P v1.5 data portal

### ISSUE THREAD

**Issue by EliVolsch** [kartoza/IGRAC-GGIS#722]
Priority: Medium

Good afternoon,

I would like to know if it is possible to update the G3P v1.5 data portal (https://ggis.un-igrac.org/view/G3P/) with a new NetCDF file for GWSA values. If so, I will send you the new file. Also, a couple of changes in the text would be necessary (which I can provide after receiving your confirmation). Thanks.

Kind regards,

Claudia

**Comment by clau1313** [comment]
Hello there, is there any update about this?

**Comment by meomancer** [comment]
Hi @clau1313 
The geonode does not provide the netcdf file, so i think we can update and upload it directly to geoserver
Please provide the files and information

**Comment by meomancer** [comment]
> Also, it is possible for us to change the metadata by ourselves? https://ggis.un-igrac.org/maps/2241/metadata_detail

Yes, you can change here: https://ggis.un-igrac.org/maps/2241/metadata

**Comment by EliVolsch** [comment]
> Can you also please:
> 
> Delete the existing layers: GWSA, GWSA_LA_grav, GWSA_LA_hyd Remove the GGMN layer from the panel Remove the words "set1 v1" from the layer "G3P Aquifers set1 v1" but please keep the layer In the title of the portal, please replace "v1.5 data" to "V1.12"
> 
> Also, it is possible for us to change the metadata by ourselves? https://ggis.un-igrac.org/maps/2241/metadata_detail
> 
> Thanks!

Hi @clau1313  You should be able to preform all of the task listed here, with the relevant permissions you account has, directly from the GeoNode portal. Can you please confirm if you are able to delete a layer. If not  @meomancer can update the permissions on your profile. 

Should you have any difficulty @seabilwe can provide guidance on how to perform the tasks.

**Comment by meomancer** [comment]
Hi @clau1313 
I'm sorry i missed your wetransfer and it is expired now
Could you recreate it?

Thank you

**Comment by clau1313** [comment]
hi @meomancer, I just sent you a new link.

**Comment by meomancer** [comment]
Hi @clau1313 
I have uploaded and create the layer based on above url:
https://ggis.un-igrac.org/catalogue/#/dataset/2520

Please check and update the detail

After that we can add it to the map and remove the old gwsa

**Comment by clau1313** [comment]
Hi @meomancer, the layer looks fine, and I updated the metadata. I had to unchecked a box that said "has time", otherwise it wouldn't let me update the metadata, but I don't know what this means. Please let me know when you add the layer to the G3P map viewer. Thanks!

**Comment by clau1313** [comment]
hi @meomancer, any update? Also, why is this taking so long? Is there any additional issue we are not aware of?

**Comment by robertohuru** [comment]
@clau1313 the data has been updated. Please check.

**Comment by clau1313** [comment]
thanks @robertohuru. Just to confirm, you used the last version of G3P that I sent a while ago, right? (version 12, it's indicated in the name of the file).

It is working but when you press play it takes quite a while to load the layer (like 10 seconds), as you can see in this screenshot, the map appears empty (first I changed the date to somewhere in the middle on the series, and then I click play).

![Image](https://github.com/user-attachments/assets/6b7d6e23-d1d3-46f9-a811-4dd3a2b63773)

Also, can it start by default at the beginning of the series and not at the end (as shown below)? so the user can just click play to see the changes?

![Image](https://github.com/user-attachments/assets/afd9bb1a-88ec-4bd8-87d5-aa090b871811)

Thanks!

**Comment by robertohuru** [comment]
> thanks [@robertohuru](https://github.com/robertohuru). Just to confirm, you used the last version of G3P that I sent a while ago, right? (version 12, it's indicated in the name of the file).

@clau1313 Yes, this is the latest file which shared. It runs from 2002 April until 2023 September.
> 
> It is working but when you press play it takes quite a while to load the layer (like 10 seconds), as you can see in this screenshot, the map appears empty (first I changed the date to somewhere in the middle on the series, and then I click play).

The speed of loading the layer depends on how fast the Geoserver serves the tiles. Since it is the first time loading, the tiles are cached. I hope that the subsequent loads whill be faster since they tiles will be fetched the web cache.

Please note that there are some missing timesteps in the series which will result in empty map. The total number of months is 225 instead 257 (difference in months between the two start and end dates).

> 
> Also, can it start by default at the beginning of the series and not at the end (as shown below)? so the user can just click play to see the changes?

Yes, by default the layer loads on a set timestamp which can be changed in the timeline and persisted after clicking the "Save" button. You can also use the timeline to change the animation step and duration.

**Comment by clau1313** [comment]
Hi @robertohuru , thanks. I tried to change the timer to the beginning of the series but it didn't work (I believe I followed your instructions). Can you please check it? thanks.

**Comment by robertohuru** [comment]
Hi @clau1313, I think I have got hold of the issue. The timeline selector was sending a wrong time to the GeoServer which was outside the accepted domain. I have fixed it by ignoring `hh:mm:ss` in the payload received by GeoServer. Please have a look again.

**Comment by clau1313** [comment]
hi @robertohuru, it still starts at the end of the series:

![Image](https://github.com/user-attachments/assets/ba73812a-ad60-4ed9-9ce9-d68e3de64d5d)

I am using this link: https://ggis.un-igrac.org/catalogue/#/dataset/2520

**Comment by robertohuru** [comment]
hi @clau1313 seems like we have been looking at different links. I am using the view

https://ggis.un-igrac.org/view/G3P/

**Comment by robertohuru** [comment]
@clau1313 that is a property of the timeline settings which can be changed by the user in the timeline settings. Currently, there is no step by month. 

<img width="307" alt="Image" src="https://github.com/user-attachments/assets/30d64621-d5ff-444a-83fd-fb337785fbec" />

@meomancer is this something you can customize in the timeline settings to allow step by month and then persist it on save?

**Comment by meomancer** [comment]
Hi @robertohuru @clau1313 
Let me check if it is able to do animation step for monthly this friday

**Comment by meomancer** [comment]
Hi @robertohuru Could we maybe do "30 animation step" and "Day" for now?
As i try to find time to fix the "month" selection

**Comment by robertohuru** [comment]
Hi @meomancer, that is okay. Can this configuration be stored/persisted for the dataset?

**Comment by meomancer** [comment]
Hi @robertohuru @clau1313 
I have fixed it
We need to turn on snap to guide layer
And after that, we save the map

![Image](https://github.com/user-attachments/assets/9252e90d-ef8c-4b1e-9c41-3c534e31813d)

Could you check it?

**Comment by clau1313** [comment]
hi @meomancer @robertohuru, thanks, I see that now the timesteps are changing monthly. However, the map itself doesn't change. I get the idea that it's loading very slow. Could that be an issue from my side?

**Comment by robertohuru** [comment]
@clau1313 this is because of the frame duration under the playback settings. This is currently defaulting to 2 seconds which for sure is very little being that the time it take to fetch the tiles from the server can be longer because of network differences. I would recommend increasing it to 5seconds.

@meomancer please can you set the default to 5seconds. @clau1313 you can also play with this parameter and recommend another frame duration.

**Comment by clau1313** [comment]
hi @meomancer, any update?

**Comment by clau1313** [comment]
hi @meomancer, @EliVolsch is there any update? why is this taking so long?

**Comment by meomancer** [comment]
Yes @clau1313 
That is correct

**Comment by clau1313** [comment]
ok @meomancer thanks!
